### ISSUE TITLE
Copy files using SCP rather than SFTP

### ISSUE THREAD

**Issue by shawark** [instaclustr/supportCenter#42]
Agent collector uses SFTP to copy files https://github.com/instaclustr/supportCenter/blob/master/agent/collector/agent.go

This works fine for most of the cases as SFTP is enabled by default on SSH. However, it has some limitations that may not work well on some customer environment setup.

* SFTP can be disabled by the customer for security concern
```
#Subsystem sftp /usr/lib/openssh/sftp-server
```
* or it may be restricted from getting any Cassandra configs
```
ChrootDirectory /path
```
* Also I have a user case that the SSH user need to use `sudo cp` to get Cassandra files (which is a legit requirement for their own security reasons). However SFTP does not to run with privileged mode

Plus, it is an extra requirement we should document in README that SFTP is required.

Therefore SCP might be preferred over SFTP to get files and with more expected behavior.

**Comment by shawark** [comment]
Having said that, for the user case I mentioned

> Also I have a user case that the SSH user need to use sudo cp to get Cassandra files (which is a legit requirement for their own security reasons). However SFTP does not to run with privileged mode

SCP also cannot run as `sudo` to copy files from Cassandra directories directly either.

How the currently consulting script works around this
- SSH user runs `sudo cp` to copy the files to a temporary location
- then the files can be `scp`

Not sure what's the best direction to go
- do the same as the consulting script. 1st copy to a temporary location and then `scp`
- Use rsync as it can do in one line `rsync --rsync-path="sudo rsync" <FILE> USER@SERVER:/var/log/cassandra`. However we need add requirement for `rsync`

**Comment by benbromhead** [comment]
Ok so my suggested solution here would be to allow the user to define a script that runs before collecition. E.g. a `pre_execute` script or something. This gives us a very flexible stop-gap measure to handle any odd cases the customer has (such as a requirement to user sudo).

Let's also look at using scp, or even just compress then pipe over the ssh tunnels stdin / stdout itself as a fallback.

**Comment by shawark** [comment]
Yeah, in that regards, supportcenter need be able to define `CP`, `NODETOOL`, `DU` or else, so we can customise
